ecs-agent/wsclient/mock/utils/utils.go

```diff
@@ -0,0 +1,67 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package utils
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+// GetMockServer returns a mock websocket server that can be started up as TLS or not.
+// TODO replace with gomock
```

Comment:
Is this TODO doable within this PR?

Reply:
This mockServer is heavily used in all existing test cases. Maybe it can be picked up in a separate PR. But it would need good amount of rewrite.